Content.Server/GameObjects/Components/Interactable/MatchstickComponent.cs

```diff
@@ -0,0 +1,112 @@
+﻿#nullable enable
+using System.Threading.Tasks;
+using Content.Shared.Audio;
+using Content.Shared.GameObjects.Components;
+using Content.Shared.Interfaces.GameObjects.Components;
+using Robust.Server.GameObjects;
+using Robust.Server.GameObjects.EntitySystems;
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameObjects.Components.Timers;
+using Robust.Shared.GameObjects.Systems;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Serialization;
+using Robust.Shared.ViewVariables;
+
+namespace Content.Server.GameObjects.Components.Interactable
+{
+    [RegisterComponent]
+    [ComponentReference(typeof(IHotItem))]
+    public class MatchstickComponent : Component, IHotItem, IInteractUsing
+    {
+        public override string Name => "Matchstick";
+
+        private SharedBurningStates _currentState;
+
+        /// <summary>
+        /// How long will matchstick last in seconds.
+        /// </summary>
+        [ViewVariables(VVAccess.ReadOnly)] private int _duration;
+
+        /// <summary>
+        /// Sound played when you ignite the matchstick.
+        /// </summary>
+        private string? _igniteSound;
+
+        /// <summary>
+        /// Point light component.
+        /// </summary>
+        private PointLightComponent? _pointLightComponent;
+
+        /// <summary>
+        /// Current state to matchstick. Can be <code>Unlit</code>, <code>Lit</code> or <code>Burnt</code>.
+        /// </summary>
+        [ViewVariables]
+        public SharedBurningStates CurrentState
+        {
+            get => _currentState;
+            private set
+            {
+                _currentState = value;
+
+                if (_pointLightComponent != null)
+                {
+                    _pointLightComponent.Enabled = _currentState == SharedBurningStates.Lit;
+                }
+
+                if (Owner.TryGetComponent(out AppearanceComponent? appearance))
+                {
+                    appearance.SetData(SmokingVisuals.Smoking, _currentState);
+                }
+            }
+        }
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+
+            serializer.DataField(ref _duration, "duration", 10);
+            serializer.DataField(ref _igniteSound, "igniteSound", null);
+        }
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            CurrentState = SharedBurningStates.Unlit;
+
+            Owner.TryGetComponent(out _pointLightComponent);
+        }
+
+        public bool IsCurrentlyHot()
+        {
+            return CurrentState == SharedBurningStates.Lit;
+        }
+
+        public void Ignite(IEntity user)
+        {
+            // Play Sound
+            if (!string.IsNullOrEmpty(_igniteSound))
+            {
+                EntitySystem.Get<AudioSystem>().PlayFromEntity(_igniteSound, user,
```

Comment:
play it from the matchstick imo
```suggestion
                EntitySystem.Get<AudioSystem>().PlayFromEntity(_igniteSound, Owner,
```